internal/generator/pedersen/template/pedersen.go.tmpl

```diff
@@ -2,62 +2,68 @@ import (
     "crypto/rand"
     "fmt"
     "github.com/consensys/gnark-crypto/ecc"
-    "github.com/consensys/gnark-crypto/ecc/{{.Name}}"
+    curve "github.com/consensys/gnark-crypto/ecc/{{.Name}}"
     "github.com/consensys/gnark-crypto/ecc/{{.Name}}/fr"
+	"io"
     "math/big"
 )
 
-// Key for proof and verification
-type Key struct {
-    g             {{.CurvePackage}}.G2Affine // TODO @tabaie: does this really have to be randomized?
-    gRootSigmaNeg {{.CurvePackage}}.G2Affine //gRootSigmaNeg = g^{-1/σ}
-    basis         []{{.CurvePackage}}.G1Affine
-    basisExpSigma []{{.CurvePackage}}.G1Affine
+// ProvingKey for committing and proofs of knowledge
+type ProvingKey struct {
+	basis         []curve.G1Affine
+	basisExpSigma []curve.G1Affine
 }
 
-func randomOnG2() ({{.CurvePackage}}.G2Affine, error) { // TODO: Add to G2.go?
-    gBytes := make([]byte, fr.Bytes)
-    if _, err := rand.Read(gBytes); err != nil {
-        return {{.CurvePackage}}.G2Affine{}, err
+type VerifyingKey struct {
+    g             curve.G2Affine // TODO @tabaie: does this really have to be randomized?
+    gRootSigmaNeg curve.G2Affine //gRootSigmaNeg = g^{-1/σ}
+}
+
+func randomFrSizedBytes() ([]byte, error) {
+    res := make([]byte, fr.Bytes)
+    _, err := rand.Read(res)
+    return res, err
+}
+
+func randomOnG2() (curve.G2Affine, error) { // TODO: Add to G2.go?
+	if gBytes, err := randomFrSizedBytes(); err == nil {
```

Comment:
this is me nit picking but, the usual pattern we see in go code bases is 
if err != nil {return error} --> (else) return result.
(and not the opposite :p )

Reply:
I thought since I was using both conditions it would be nice to consider the full half of the glass first for a change. 😁